clcache.py

```diff
@@ -232,6 +232,9 @@ def getManifestHash(compilerBinary, commandLine, sourceFile):
         # One of the few exceptions to this rule is the /MP switch, which only
         # defines how many compiler processes are running simultaneusly.
         commandLine = [arg for arg in commandLine if not arg.startswith("/MP")]
+        baseDir = os.environ.get('CLCACHE_BASEDIR')
+        if baseDir:
+            commandLine = [arg.replace(baseDir, BASEDIR_REPLACEMENT) for arg in commandLine]
```

Comment:
Yes, I don't think it is correct to apply the replacement blindly. I can instead add a list of parameters and apply the replacement only to those.

In that case, which arguments should be in the white list? I had a look here (https://msdn.microsoft.com/en-us/library/fwkeyyhe.aspx), these are the arguments that seem good candidates:

 - "/I" https://msdn.microsoft.com/en-us/library/73f9s62w.aspx
 - "/AI" https://msdn.microsoft.com/en-us/library/x1x72k9t.aspx
 - "/FU" https://msdn.microsoft.com/en-us/library/81ex1b0a.aspx
 - "/Yu" https://msdn.microsoft.com/en-us/library/z0atkd6c.aspx